lua/opencode/util.lua

```diff
@@ -347,4 +347,30 @@ function M.parse_dot_args(args_str)
   return result
 end
 
+--- Check if prompt is allowed via guard callback
+--- @param guard_callback? function
+--- @return boolean allowed
+--- @return string|nil error_message
+function M.check_prompt_allowed(guard_callback)
```

Comment:
I would also allow to modify the text of the prompt ?

Even the context object where you can remove files or anything.

It would be more flexible. If the function see a file in the context it should not sent it could call context.remove_file (the function is not there yet but would be trivial to add).

What do you think ?